util/srconv.c

```diff
@@ -219,20 +219,20 @@ static int32_t srconv(CSOUND *csound, int32_t argc, char **argv)
     char        outformch = 's';
     unsigned    outbufsiz = 0U;
     SNDFILE     *outfd = NULL;
-    OPARMS      O;
+   const OPARMS *     O;
```

Comment:
This looks odd as it's declaring a pointer then assigning ->outformat a few lines down. I suspect this and the other utils (which are memcpy'ing OPARMS) may have problems as I don't see how they would work when OPARMS was made read-only now. I see in some files there's a ->Calloc of sizeof OPARMS.

Reply:
The oparms here and in the other utilities is used only locally to pass options from one function to another and not to the rest of Csound. So it takes in the options from Csound (which are readonly and copies them locally, making some local changes.

Reply:
Yes, I should have allocated and copied this but I fixed it now.
The oparms here and in the other utilities is used only locally to pass options from one function to another and not to the rest of Csound. So it takes in the options from Csound (which are readonly and copies them locally, making some local changes.

Reply:
I also noted some unused OPARMS code in another file and removed it. mkdb.c which we don't compile had also been the subject of a bad replacement.